db/migrate/20170822132949_create_subscriptions.rb

```diff
@@ -0,0 +1,21 @@
+class CreateSubscriptions < ActiveRecord::Migration[5.1]
+
+  def change
+    create_table :subscriptions do |t|
+      t.string :regcode, null: false
+      t.string :name, null: false
+      t.string :kind, null: false
+      t.string :status, null: false
+
+      t.datetime :starts_at
+      t.datetime :expires_at
+
+      t.integer :system_limit, null: false
+      t.integer :systems_count, null: false
+      t.integer :virtual_count
+
+      t.timestamps
+    end
```

Comment:
Are you sure that you don't need any indices on subscriptions? The base controller at least searches by `regcode`.

Reply:
Yep, I'll throw in an index.